server/routes/record-attendance/handlers/attendanceList.test.ts

```diff
@@ -187,11 +190,73 @@ describe('Route Handlers - Attendance List', () => {
       expect(res.redirect).toBeCalledWith('attendance-list')
     })
 
-    it("shouldn't update attendance when no pirsoners have been selected", async () => {
+    it("shouldn't update attendance when no prisoners have been selected", async () => {
       await handler.ATTENDED(req, res)
 
       expect(activitiesService.updateAttendances).toBeCalledTimes(0)
       expect(res.redirect).toBeCalledWith('attendance-list')
     })
+
+    it('non attendance should be redirected to the non attendance page', async () => {
+      req = {
+        params: { id: 1 },
+        session: {
+          notAttendedJourney: {},
+        },
+        body: {
+          selectedAttendances: ['1-ABC123'],
+        },
+      } as unknown as Request
+
+      when(activitiesService.getScheduledActivity)
+        .calledWith(1, res.locals.user)
+        .mockResolvedValue({
+          id: 1,
+          date: format(new Date(), '2022-12-08'),
+          startTime: '10:00',
+          endTime: '11:00',
+          activitySchedule: {
+            activity: { summary: 'Maths level 1' },
+            internalLocation: { description: 'Houseblock 1' },
+          },
+          attendances: [
+            { prisonerNumber: 'ABC123', status: 'WAITING' },
+            { prisonerNumber: 'ABC321', status: 'COMPLETED', attendanceReason: { code: 'ATT' } },
```

Comment:
WAITING / COMPLETED look right to me at status codes, but the reason code ATT? Should it be ATTENDED in the UI side?

Reply:
Corrected